server/http_status.go

```diff
@@ -34,50 +30,32 @@ var once sync.Once
 const defaultStatusAddr = ":10080"
 
 func (s *Server) startStatusHTTP() {
-	// prepare region cache for region http server
-	pdClient, err := s.getPdClient()
-	if err != nil {
-		log.Fatal(err)
-	}
-
 	once.Do(func() {
-		go s.startHTTPServer(pdClient)
+		go s.startHTTPServer()
 	})
 }
 
-func (s *Server) getPdClient() (pd.Client, error) {
-	if s.cfg.Store != "tikv" {
-		return nil, nil
-	}
-	path := fmt.Sprintf("%s://%s", s.cfg.Store, s.cfg.StorePath)
-	etcdAddrs, err := tikv.ParseEtcdAddr(path)
-	if err != nil {
-		return nil, err
-	}
-	client, err := pd.NewClient(etcdAddrs)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	return client, nil
-}
-
-func (s *Server) startHTTPServer(pdClient pd.Client) {
+func (s *Server) startHTTPServer() {
 	router := mux.NewRouter()
 	router.HandleFunc("/status", s.handleStatus)
 	// HTTP path for prometheus.
 	router.Handle("/metrics", prometheus.Handler())
-
-	// HTTP path for regions
-	router.Handle("/tables/{db}/{table}/regions", s.newTableRegionsHandler(pdClient))
-	router.Handle("/regions/{regionID}", s.newRegionHandler(pdClient))
-
+	tikvHandler, err := s.newRegionHandler()
+	if err == nil {
```

Comment:
Since the error won't be nil when the store is not `tikv`, I think it's more appropriate to log an info here @siddontang .